docs/content/docs.md

```diff
@@ -2110,33 +2110,43 @@ there were IO errors`.
 ### --fast-list ###
 
 When doing anything which involves a directory listing (e.g. `sync`,
-`copy`, `ls` - in fact nearly every command), rclone normally lists a
-directory and processes it before using more directory lists to
-process any subdirectories.  This can be parallelised and works very
-quickly using the least amount of memory.
+`copy`, `ls` - in fact nearly every command), the basic strategy
+of rclone is to list one directory and processes it before using more
+directory lists to process any subdirectories.  This can be parallelised
+and works very quickly using the least amount of memory.
 
-However, some remotes have a way of listing all files beneath a
-directory in one (or a small number) of transactions.  These tend to
-be the bucket-based remotes (e.g. S3, B2, GCS, Swift).
-
-If you use the `--fast-list` flag then rclone will use this method for
-listing directories.  This will have the following consequences for
-the listing:
+However, some backends provide the support for an alternative strategy,
+where all files beneath a directory are listed in one (or a small number)
+of transactions. This will have the following consequences for the listing:
 
   * It **will** use fewer transactions (important if you pay for them)
-  * It **will** use more memory.  Rclone has to load the whole listing into memory.
+  * It **will** use more memory because rclone has to load the whole listing into memory
```

Comment:
`ListR` when used by `rclone ls` doesn't use more memory.

You need to explicitly provide the `--fast-list` flag to allow rclone to use more memory.

Reply:
> ListR when used by rclone ls doesn't use more memory.

Right, because it is creating the resulting file objects for the entire list regardless, summing up to the same amount of memory? However, when submitting a request for the entire tree the response xml will surely be bigger, i.e. using more memory, than if requesting one directory or whatever at a time, but I guess that is small in comparison.

Reply:
If you use `--fast-list` in `rclone sync` then rclone will load in every object in the listing tree into memory which might be millions of objects which uses GB of memory.

However just using `ListR` on its own doesn't use significantly more memory as it will be getting the objects in chunks of 1000 (s3 for example) and dealing with them before asking for the next chunk.

`rclone ls` will call `ListR` and output the objects to stdout as they come in, without storing them.